### ISSUE TITLE
hyperparameters during inference

### ISSUE THREAD

**Issue by tonytan48** [grammarly/gector#22]
Hi,

Thank you for the novel idea and great work.  I was trying to learn and reproduce the inference result with the provided pretrained roberta and XLNet model. However, referring to the paper, I had some confusion on hyperparameter during inference. In the appendix does the confidence bias refer to --additional_confidence? I tried to reproduce this result by XLNet with --min_error_probability 0.66 --additional_confidence 0.35:

CUDA_VISIBLE_DEVICES='0' python predict.py --model_path pretrained_models/xlnet_0_gector.th --vocab_path data/output_vocabulary --input_file 'data/wi+locness/m2/ABCN.dev.gold.bea19.original' --output_file output/xlnet.ABCN.dev.bea19.corr.min_error_0.66.add_conf_0.35.txt --transformer_model xlnet --special_tokens_fix 0 --min_error_probability 0.66 --additional_confidence 0.35
Produced overall corrections: 2438

However, with the XLNet model on BEA19 dev set I only obtained the following result. Could you help to provide the inference command for roberta and XLNet released pretrained model? It will be very helpful in the appendix as well

 Span-Based Correction 
TP	FP	FN	Prec	Rec	F0.5
2523	1295	4938	0.6608	0.3382	0.5549
 
 Span-Based Detection 
TP	FP	FN	Prec	Rec	F0.5
2851	981	4781	0.744	0.3736	0.6209
 

Token-Based Detection 
TP	FP	FN	Prec	Rec	F0.5
3415	802	5581	0.8098	0.3796	0.6602

Thank you!

**Comment by skurzhanskyi** [comment]
Hi @tonytan48 
It looks like you use proper hyperparameters. On the BEA19 dev set, models usually have a significantly lower ERRANT score than on test set. Please check the scores on the BEA19 test set.

**Comment by tonytan48** [comment]
Thank you for the prompt clarification. I have a quick question on obtaining the gold edits of ABCN test data. It seems it's not included in the wi+locness folder. May I know if you can share the m2 file for searching and tuning the hyperparameters?

**Comment by gurunath-p** [comment]
@tonytan48  For the W&I+LOCNESS dataset(ABCN), the test set is not openly released. On the BEA website, you are given a file containing the source sentences and you have to upload outputs of your model to their site, which will provide you with the scores of your model. For more details, refer BEA site [here](https://www.cl.cam.ac.uk/research/nl/bea2019st/#eval).

**Comment by skurzhanskyi** [comment]
@gurunath-p Thanks for clarifying this.

**Comment by skurzhanskyi** [comment]
@tonytan48 That's the idea - to make you not to fine-tune for the test set, so the scores there can be fairer.

**Comment by tonytan48** [comment]
> @tonytan48 For the W&I+LOCNESS dataset(ABCN), the test set is not openly released. On the BEA website, you are given a file containing the source sentences and you have to upload outputs of your model for these source sentences to their [Codalab site](https://competitions.codalab.org/competitions/20228#learn_the_details), which will provide you with the scores of your model. For more details, refer BEA site [here](https://www.cl.cam.ac.uk/research/nl/bea2019st/#eval).

Thanks for the clarification.
